pkg/network/deviceinfo/deviceinfo.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package deviceinfo
+
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+)
+
+func MapBindingPluginNetworkNameToDeviceInfo(networks []v1.Network, interfaces []v1.Interface,
+	networkStatusAnnotationValue string, bindingPlugins map[string]v1.InterfaceBindingPlugin) (map[string]*networkv1.DeviceInfo, error) {
+	return mapNetworkNameToDeviceInfo(networks, networkStatusAnnotationValue,
+		filterDeviceInfoBindingPluginEntries(interfaces, bindingPlugins))
+}
+
+func mapNetworkNameToDeviceInfo(networks []v1.Network,
+	networkStatusAnnotationValue string,
+	interfaces []v1.Interface) (map[string]*networkv1.DeviceInfo, error) {
+	multusInterfaceNameToNetworkStatus, err := mapMultusInterfaceNameToNetworkStatus(networkStatusAnnotationValue)
+	if err != nil {
+		return nil, err
+	}
+	networkNameScheme := namescheme.CreateNetworkNameSchemeByPodNetworkStatus(networks, multusInterfaceNameToNetworkStatus)
+
+	networkDeviceInfo := map[string]*networkv1.DeviceInfo{}
+	for _, iface := range interfaces {
+		multusInterfaceName := networkNameScheme[iface.Name]
+		networkStatusEntry, exist := multusInterfaceNameToNetworkStatus[multusInterfaceName]
+		if !exist {
+			continue // The interface is not plugged yet
+		}
+		networkDeviceInfo[iface.Name] = networkStatusEntry.DeviceInfo
```

Comment:
I find `continue` messy.
Perhaps you will agree to a direct approach:
```go
if networkStatusEntry, exist := multusInterfaceNameToNetworkStatus[multusInterfaceName]; exist {
    networkDeviceInfo[iface.Name] = networkStatusEntry.DeviceInfo
}
```

Reply:
done.